lib/Configuration.php

```diff
@@ -316,23 +313,17 @@ public function saveConfiguration() {
 	}
 
 	public function isDefault() {
-		$diff = \array_diff_assoc($this->getTranslatedConfig(), $this->getDefaults());
-		return \count($diff) === 0;
-	}
-
-	/**
-	 * @param string $varName
-	 * @return array|string
-	 */
-	protected function getMultiLine($varName) {
-		$value = $this->getValue($varName);
-		if (empty($value)) {
-			$value = '';
-		} else {
-			$value = \preg_split('/\r\n|\r|\n/', $value);
+		$c = $this->getTranslatedConfig();
+		// multiple values means the config is not default
+		foreach ($this->arrayConfigKeys as $arrayConfigKey) {
```

Comment:
I see some problems with the default values:
* Why the "default" value in the `$data` attribute are different than the ones fetched from the `getDefaults` function? I'd prefer to remove that information and use something like `$this->data = $this->getDefaults()` in the constructor in order to have only one source of information. Then we can overwrite the values accordingly to the specific DB information.
* The `arrayConfigKeys` shouldn't be needed. You have the actual data in `$this->data` and also the default values. Just traverse the array checking each value. Most of the time, the first value (ldapHost) will be different, so it will be a fast check. No need to store an additional array or perform heavy operations.

Reply:
removed